scripts/json_overview_image_info.py

```diff
@@ -31,6 +32,21 @@
                 image_info["profiles"][device_id]["images"][0]
             )
 
+
+output["default_packages"] = run(
+    [
+        "make",
+        "--no-print-directory",
+        "-C",
+        f"target/linux/{output['target'].split('/')[0]}",
+        "val.DEFAULT_PACKAGES",
+    ],
+    capture_output=True,
+    check=True,
+    env=environ.copy().update({"TOPDIR": Path().cwd()}),
+    text=True,
+).stdout.split()
```

Comment:
I don't really understand what this does, but to me it looks like it's just parsing the files for Make variables?!

Reply:
Correct

Reply:
Isn't there a better way? That's prone to overlook stuff or interpret it differently than Make will do ...
Does it account for DEFAULT_PACKAGES in include/target.mk and the different DEVICE_TYPE values?
Does it account for DEFAULT_PACKAGES in subtarget/*.mk files?

I think adding this when it possibly works different from what actually adds the packages in the code will actually be worse than not including it at all.

Reply:
From my understanding it should do all that magic.

My result with `mediatek/mt7623`

```
user@dawn:~/src/openwrt/openwrt$ make -C target/linux/mediatek/ val.DEFAULT_PACKAGES
make: Entering directory '/home/user/src/openwrt/openwrt/target/linux/mediatek'
base-files libc libgcc busybox dropbear mtd uci opkg netifd fstools uclient-fetch logd urandom-seed urngd kmod-leds-gpio kmod-gpio-button-hotplug dnsmasq iptables ip6tables ppp ppp-mod-pppoe firewall odhcpd-ipv6only odhcp6c kmod-ipt-offload kmod-leds-gpio kmod-gpio-button-hotplug
make: Leaving directory '/home/user/src/openwrt/openwrt/target/linux/mediatek'
```

And with `mediatek/mt7622`

```
user@dawn:~/src/openwrt/openwrt$ make -C target/linux/mediatek/ val.DEFAULT_PACKAGES
make: Entering directory '/home/user/src/openwrt/openwrt/target/linux/mediatek'
base-files libc libgcc busybox dropbear mtd uci opkg netifd fstools uclient-fetch logd urandom-seed urngd kmod-leds-gpio kmod-gpio-button-hotplug kmod-mt7615e kmod-mt7615-firmware wpad-basic dnsmasq iptables ip6tables ppp ppp-mod-pppoe firewall odhcpd-ipv6only odhcp6c kmod-ipt-offload kmod-leds-gpio kmod-gpio-button-hotplug
make: Leaving directory '/home/user/src/openwrt/openwrt/target/linux/mediatek'
```

E.g. `kmod-mt7615e` is only included for `mt7622` and therefore I think that `subtarget/*.mk` is considered.

Also it includes a bunch of PPP related stuff which is added for the type router.